src/wp-includes/fonts.php

```diff
@@ -131,7 +131,44 @@ function wp_get_font_dir() {
 	 *
 	 * @param array $defaults The original fonts directory data.
 	 */
-	return apply_filters( 'font_dir', $defaults );
+	$font_dir = apply_filters( 'font_dir', $default_dir );
+
+	wp_mkdir_p( $font_dir['path'] );
+
+	if ( ! is_dir( $font_dir['path'] ) || ! wp_is_writable( $font_dir['path'] ) ) {
+
+		// If the default directory was filtered it uses the filtered values and doesnt fall back to the wp-content/uploads/fonts.
+		if ( $font_dir !== $default_dir ) {
+			$font_dir['error'] = sprintf(
+				/* translators: %s: Directory path. */
+				__( 'Unable to create directory %s. Is its parent directory writable by the server?' ),
+				esc_html( str_replace( ABSPATH, '', $font_dir['path'] ) )
+			);
```

Comment:
Note that there are a few places where `wp_get_font_dir` is used that we'll need to check to see if the `error` is populated and handle it accordingly:

- https://github.com/WordPress/wordpress-develop/blob/17415a747166e0c90bcf8417d689ad9c556fb90b/src/wp-includes/fonts.php#L178
- https://github.com/WordPress/wordpress-develop/blob/17415a747166e0c90bcf8417d689ad9c556fb90b/src/wp-includes/rest-api/endpoints/class-wp-rest-font-faces-controller.php#L869
- https://github.com/WordPress/wordpress-develop/blob/17415a747166e0c90bcf8417d689ad9c556fb90b/src/wp-includes/rest-api/endpoints/class-wp-rest-font-faces-controller.php#L869